src/build.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+import path from 'path';
+import {createWriteStream} from 'fs';
+import streamToPromise from 'stream-to-promise';
+
+import * as fs from './util/promised-fs';
+import {zipDir} from './util/zip-dir';
+
+
+export default function build({sourceDir, buildDir}: Object) {
```

Comment:
> having some integration testing even if the core tests in the tree are just unit tests

That's exactly what I'd like to avoid though. If we start to rely on integration tests to solve the problems I described then we will make the suite slow. A slow test suite is a slow death. I've seen it on every project I've worked on. When is the last time you were excited about running all of addons-server's tests?

To be clear, I think we should eventually add integration tests but only a very few -- i.e. to act more as "smoke tests" rather than tests that address the integration of components.

> I think it would be good if it didn't require additional syntax

It will definitely require additional syntax because some types can't be inferred (it's a limitation of JavaScript). I did some searching and I see that comment-based syntax is supported but not really documented: http://flowtype.org/blog/2015/02/20/Flow-Comments.html At first it seemed like a good idea but adding a transpilation step is not a problem for us. I fear that relying on comments will just make the code harder to read.